rllib/env/multi_agent_episode.py

```diff
@@ -967,35 +997,51 @@ def _generate_single_agent_episode(
                 None
                 if actions is None
                 else self._get_single_agent_data(
-                    agent_id, actions, start_index=1, shift=-1
+                    #    agent_id, actions, start_index=1, shift=-1
+                    agent_id,
+                    actions,
+                    use_global_to_to_local_t=False,
                 )
             )
+
+            # Rewards are special in multi-agent scenarios, as agents could receive a
+            # reward even though they did not get an observation at a specific timestep.
             agent_rewards = (
                 None
                 if rewards is None
                 else self._get_single_agent_data(
-                    agent_id, rewards, start_index=1, shift=-1
+                    agent_id,
+                    rewards,
+                    use_global_to_to_local_t=False,
+                    start_index=1,
```

Comment:
We can get rid here of the start index and the shift. No need for this when we use `use_global_t_to_local_t=False` 

Also fix the bug in the naming.